include/mathfu/matrix.h

```diff
@@ -1360,6 +1388,36 @@ static inline Matrix<T, 4, 4> LookAtHelper(const Vector<T, 3>& at,
 }
 /// @endcond
 
+/// @cond MATHFU_INTERNAL
+///
+template <class T>
+static inline bool UnProjectHelper(const Vector<T, 3>  windowCoord, 
+                                   const Matrix<T, 4, 4> modelView,
+                                   const Matrix<T, 4, 4> projection,
+                                   const float windowWidth,
+                                   const float windowHeight,
+                                   Vector<T, 3>& result) {
+  if (windowCoord.z()<0 || windowCoord.z()>1) {
```

Comment:
style nit: whitespace between comparison operators ...

if (window_coord.z() < 0 || window_coord.z() > 1) {

also we have some fussy compilers so you should probably static_cast<T> the constants in here.